crates/emitter/src/emitter_scope.rs

```diff
@@ -285,4 +307,10 @@ impl EmitterScopeStack {
 
         NameLocation::Dynamic
     }
+
+    pub fn current_index(&mut self) -> EmitterScopeIndex {
```

Comment:
I guess it's better calling this a `depth` instead? (`EmitterScopeDepth`, and `current_depth`)
given that, same index doesn't point the same value once the scope gets popped from the stack.
so the name should be more associated with enter/leave.